### PR TITLE
New release 0.1.0

### PR DESCRIPTION
**Pull request by JunCEEE**
- The architecture was upgraded with the [design](https://github.com/PaNOSC-ViNYL/libpyvinyl/raw/wiki_images/wiki/example_usage.png?raw=true) to support multiple formats better. 
- `BaseData` class was introduced.
- `DataCollections` class as a thin layer for data aggregating. 
- `BaseCalcutor` class is renewed with `BaseData` class.
- `Parameter.value` supports NumPy array.
- `Parameter.value` deals units with [pint package](https://pint.readthedocs.io/en/stable/) internally.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,33 +2,109 @@
 
 [![Build Status](https://travis-ci.com/PaNOSC-ViNYL/libpyvinyl.svg?branch=master)](https://travis-ci.com/PaNOSC-ViNYL/libpyvinyl)
```
Comment (CFGrote) [inline README.md:3]: Build status icon is still taken from travis, should be gihub actions, right?

File: README.md [hunk 2]
```diff
@@ -2,33 +2,109 @@
...
 [![Documentation Status](https://readthedocs.org/projects/libpyvinyl/badge/?version=latest)](https://libpyvinyl.readthedocs.io/en/latest/?badge=latest)
```
Comment (CFGrote) [inline README.md:4]: Can you make sure,rtd works as intended?

**Review by CFGrote — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (JunCEEE) [reply README.md:4]: Working on that.

Comment (JunCEEE) [reply README.md:4]: @CFGrote Could you grant me management access to the RTD?

File: README.md (hunk 1 above)
Comment (JunCEEE) [reply README.md]: Fixed: https://github.com/PaNOSC-ViNYL/libpyvinyl/tree/develop#readme

File: README.md (hunk 2 above)
Comment (JunCEEE) [reply README.md:4]: I checked https://libpyvinyl.readthedocs.io/en/develop/, the [DEMO](https://libpyvinyl.readthedocs.io/en/develop/include/notebooks/example-01.html#Implement-a-Calculator-that-derives-from-BaseCalculator.) seems not working. Should we aim for fixing it in PR #49?

File: libpyvinyl/BeamlinePropagator.py [hunk 3]
```diff
@@ -27,15 +27,16 @@
 
 from libpyvinyl.BaseCalculator import BaseCalculator, CalculatorParameters
 
+
```
Comment (shervin86) [inline libpyvinyl/BeamlinePropagator.py:30]: Should this file be removed?

File: libpyvinyl/Instrument.py [hunk 4]
```diff
@@ -3,26 +3,32 @@
...
         """An Instrument class
 
         :param name: The name of this instrument
         :type name: str
         :param calculators: A collection of Calculator objects.
         :type calculators: dict
         """
-        self.__name = name
-        self.__calculators = {}
+        self.__name = None
```
Comment (shervin86) [inline libpyvinyl/Instrument.py:19]: Why setting it to None when there is a setter call immediately after? Can this line simply be removed?

File: tests/integration/plusminus/HISTORY.rst [hunk 5]
```diff
@@ -0,0 +1,8 @@
+=======
```
Comment (shervin86) [inline tests/integration/plusminus/HISTORY.rst:1]: Shall we remove this file?

File: tests/integration/plusminus/LICENSE [hunk 6]
```diff
@@ -0,0 +1,22 @@
+MIT License
```
Comment (shervin86) [inline tests/integration/plusminus/LICENSE:1]: I would remove this. I think we can be fine with the libpyvinyl license

File: tests/integration/plusminus/MANIFEST.in [hunk 7]
```diff
@@ -0,0 +1,11 @@
+include CONTRIBUTING.rst
```
Comment (shervin86) [inline tests/integration/plusminus/MANIFEST.in:1]: To be removed

File: tests/integration/plusminus/Makefile [hunk 8]
```diff
@@ -0,0 +1,85 @@
+.PHONY: clean clean-test clean-pyc clean-build docs help
```
Comment (shervin86) [inline tests/integration/plusminus/Makefile:1]: To be removed?

File: tests/integration/plusminus/README.rst [hunk 9]
```diff
@@ -0,0 +1,37 @@
+=========
```
Comment (shervin86) [inline tests/integration/plusminus/README.rst:1]: To be removed

File: tests/integration/plusminus/docs/Makefile [hunk 10]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
```
Comment (shervin86) [inline tests/integration/plusminus/docs/Makefile:1]: To be removed?

**Review by shervin86 — CHANGES_REQUESTED** [review]
There are some files in the plusminus integration that in my opinion should be removed.

File: libpyvinyl/BeamlinePropagator.py (hunk 3 above)
Comment (JunCEEE) [reply libpyvinyl/BeamlinePropagator.py:30]: Sure.

File: tests/integration/plusminus/HISTORY.rst (hunk 5 above)
Comment (JunCEEE) [reply tests/integration/plusminus/HISTORY.rst:1]: Yes.

File: tests/integration/plusminus/README.rst (hunk 9 above)
Comment (JunCEEE) [reply tests/integration/plusminus/README.rst:1]: I kept it to explain what this example is doing.

File: tests/integration/plusminus/docs/Makefile (hunk 10 above)
Comment (JunCEEE) [reply tests/integration/plusminus/docs/Makefile:1]: Done

File: tests/integration/plusminus/MANIFEST.in (hunk 7 above)
Comment (JunCEEE) [reply tests/integration/plusminus/MANIFEST.in:1]: Done

**Review by shervin86 — APPROVED** [review]

**Review by CFGrote — APPROVED** [review]

**Comment by JunCEEE** [comment]
@CFGrote thanks for merging the branch. I just wonder if it's the best way to squash the commits instead of to rebase? How to track the previous developing history and pick up a specific commit if all the commits are squashed?

**Comment by CFGrote** [comment]
that's true, we should retain the history next time. maybe it's even better rebase instead of merge.

**Pull request merged**